offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/GridUtils.groovy

```diff
@@ -54,6 +58,42 @@ class GridUtils {
         filterTextField.setSizeFull()
     }
 
+    /**
+     * Provides a filter field into a header row of a grid for a given column.
+     *
+     * <p>This filter tests for the provided predicate based on textual user input.</p>
+     *
+     *
+     * @param <T> the grid bean type
+     * @param <V> the column value type
+     * @param dataProvider The grid's {@link ListDataProvider}
+     * @param column The column to add the filter to
+     * @param predicate the predicate that has to be true for a tested value and a user input String
+     * @param headerRow The {@link com.vaadin.ui.components.grid.HeaderRow} of the {@link Grid}, where the filter input field is added
+     */
+    static <T,V> void setupColumnFilter(ListDataProvider<T> dataProvider,
+                                        Grid.Column<T, V> column,
+                                        BiPredicate<V, String> predicate,
+                                        HeaderRow headerRow) {
+        TextField filterTextField = new TextField()
+        filterTextField.addValueChangeListener(event -> {
+            dataProvider.addFilter(element ->
+                predicate.test(column.getValueProvider().apply(element), filterTextField.getValue())
+            )
+        })
+        filterTextField.setValueChangeMode(ValueChangeMode.EAGER)
+        filterTextField.addStyleName(ValoTheme.TEXTFIELD_TINY)
+        String columnId = StringUtils.join(
+                StringUtils.splitByCharacterTypeCamelCase(column.id),
+                ' '
+        )
+        filterTextField.setPlaceholder("Filter by " + columnId)
+        headerRow.getCell(column).setComponent(filterTextField)
+        filterTextField.setSizeFull()
+    }
```

Comment:
can we omit the redundant code by calling the other method/extract the same behaviour into another method?

Reply:
Yes, we could extract the duplicate code.